automation-jmeter-plugins/src/main/java/automation/plugins/jmeter/functions/OAuth2Token.java

```diff
@@ -0,0 +1,178 @@
+                GoogleCredentials credentials = null;
+                try {
+                    credentials = GoogleCredentials.fromStream(new ByteArrayInputStream(saKey))
+                            .createScoped(new ArrayList<String>() {
+                                private static final long serialVersionUID = 2644360116969700916L;
+                                {
+                                    add("profile");
+                                    add("email");
+                                    add("openid");
+                                }
+                            }).createDelegated(userEmail);
+                    credentials.refreshIfExpired();
+                    authToken = credentials.getAccessToken().getTokenValue();
+                } catch (IOException e) {
+                    log.warn(e.getMessage());
+                    throw new RuntimeException(e.getMessage());
+                }
+                break;
+            case AWS:
+            case AZURE:
+                log.warn(cloudProvider.getDescription() + " not implemented");
+                throw new RuntimeException(cloudProvider.getDescription() + " not implemented");
+            default:
+                log.warn(cloudProvider.getDescription());
+                throw new RuntimeException(cloudProvider.getDescription());
+        }
+
+        if (StringUtils.isEmpty(authToken)) {
+            log.warn("Unable to obtain oauth2 token for user '{}'", userEmail);
+            throw new RuntimeException(String.format("Unable to obtain oauth2 token for user '%s'", userEmail));
+        }
+
+        String varName = null;
+
+        if (values.length > 3)
+            varName = ((CompoundVariable) values[3]).execute().trim();
+
+        if (vars != null && varName != null) {
+            vars.put(varName, authToken);
+        }
+
+        return authToken;
+    }
+
+    @Override
+    public void setParameters(Collection<CompoundVariable> parameters) throws InvalidVariableException {
+        checkParameterCount(parameters, MIN_PARAMETER_COUNT, MAX_PARAMETER_COUNT);
+        values = parameters.toArray();
+    }
+
+    @Override
+    public String getReferenceKey() {
+        return KEY;
+    }
+
+    private static String getResString(String key) {
+        ResourceBundle bundle = ResourceBundle.getBundle("messages");
+        return bundle.containsKey(key) ? bundle.getString(key) : String.format("[res_key=\"%s\"]", key);
+    }
+
+    private static byte[] getBytesFromEnvVarOrConfigPath(String env) {
+        String envVarOrConfigPath = System.getenv(env);
+        File f = new File(envVarOrConfigPath);
+        if (f.exists()) {
+            try {
+                return Files.readAllBytes(f.toPath());
+            } catch (IOException e) {
+                throw new RuntimeException(e.getMessage());
+            }
+        }
+        return envVarOrConfigPath.getBytes();
+    }
+
+    // Cloud providers
+    enum CloudProvider {
```

Comment:
Should this be moved to its own (separate) file?

Reply:
Let's just keep the `enum` In place until one more use case needs it.